src/posts/comparing-methods-for-accessing-secrets-in-vault-from-kubernetes.md

```diff
@@ -0,0 +1,285 @@
+
+|  | Secrets Engines | Auth Methods | Sync to K8s Secrets | Secret Rotation |
+| --- | --- | --- | --- | --- |
+| ArgoCD Vault Plugin | KV only | AppRole, Token, Github, Kubernetes and Userpass | Optional | Manual |
+
+## Wrapping up
+
+When considering the effort involved in the different approaches, it is important to consider your context. Are you a small, highly connected team, or are you a big enterprise with dedicated platform teams? In the former case, perhaps it doesn’t matter so much where the effort is placed. But in the latter case, it should be a goal to pay the effort on the platform team as much as possible to enable stream-aligned teams to securely access their secrets with minimal effort. After all, there will be many more stream-aligned teams than platform teams so the approach should be favourably scalable.
+
+|  | Platform Effort | Deployment Effort | Application Effort |
+| --- | --- | --- | --- |
+| Vault REST API | ■□□□□ | ■■□□□ | ■■■■■ |
+| Vault Sidecar Injector | ■■□□□ | ■■■■■ | ■■□□□ |
+| Vault CSI Driver | ■■■□□ | ■■■■□ | ■□□□□ |
+| External Secrets Operator | ■■□□□ | ■■■□□ | ■□□□□ |
+| ArgoCD Vault Plugin | ■■■□□ | ■■□□□ | ■□□□□ |
+
+When considering the features of the various approaches, a few things we encountered that helped guide the decision:
+
+1. **Third party tools**. If you manage a lot of 3rd party tools on Kubernetes, you might find it quite painful to try and use an approach that does not support synchronising with Kubernetes secrets. This is because not all tools support sane approaches for reading secrets, or if you rely on Helm, the Helm chart might not work as you intend or differ greatly from tool to tool.
+2. **Scalability**. If you work with multi-cluster setups, each cluster will need access to Vault. This can become a configuration pain if using the [Kubernetes Auth Method](https://developer.hashicorp.com/vault/docs/auth/kubernetes) with Vault, as each cluster needs its own auth engine. Sure, you can automatically add each cluster as an auth method when you create it (e.g. with Terraform), but the real pain is when using the auth method in your Kubernetes manifests as you need to provide the unique name of auth method. Suddenly your workloads need to be cluster-aware (or your deployment tool can handle that, which actually can be done using [ArgoCD ApplicationSets](https://argocd-applicationset.readthedocs.io/en/stable/Template/), for example).
+3. **Secrets rotation**. If you use dynamic secrets and there is a chance your workload could fail to authenticate once the secret expires then secrets rotation should be considered.
+4. **Multi-cloud with cross-cloud access**. Dynamic secrets should be high on the agenda so that, for example, a pod in one cloud can get dynamic credentials for an IAM account in another cloud and access some cloud resources securely. Immediately we filter out the approaches that only support key-value secrets.
```

Comment:
```suggestion
4. **Multi-cloud with cross-cloud access**. Dynamic secrets should be high on the agenda so that, for example, a pod in one cloud can get dynamic credentials for an IAM account in another cloud and access some cloud resources securely. This immediately filters out the approaches that only support key-value secrets.
```